apps/dashboard/app/(app)/apis/[apiId]/settings/components/update-ip-whitelist.tsx

```diff
@@ -0,0 +1,151 @@
+    );
+  return (
+    <form onSubmit={form.handleSubmit(onSubmit)}>
+      <SettingCard
+        className="mt-[20px] py-[19px] w-full"
+        title={
+          <div className=" flex items-center justify-start gap-2.5">
+            <span className="text-sm font-medium text-accent-12">IP Whitelist</span>
+            <Badge />
+          </div>
+        }
+        description={
+          <div className="font-normal text-[13px] max-w-[380px]">
+            Want to protect your API keys from unauthorized access? <br />
+            Upgrade to our <span className="font-bold">Enterprise plan</span> to enable IP
+            whitelisting and restrict access to trusted sources.
+          </div>
+        }
+        border="both"
+        contentWidth="w-full lg:w-[320px]"
+      >
+        {isEnabled ? (
+          <div className="flex flex-row justify-items-stretch items-center w-full gap-x-2">
+            <input type="hidden" name="workspaceId" value={api.workspaceId} />
+            <input type="hidden" name="apiId" value={api.id} />
+            <label htmlFor="ipWhitelist" className="hidden sr-only">
+              IP Whitelist
+            </label>
+            <FormField
+              control={form.control}
+              name="ipWhitelist"
+              render={({ field }) => (
+                <Textarea
+                  className="max-w-sm"
+                  {...field}
+                  autoComplete="off"
+                  placeholder={`127.0.0.1
+1.1.1.1`}
+                />
+              )}
+            />
+            <Button
+              size="lg"
+              className="rounded-lg px-2.5"
+              disabled={
+                !form.formState.isValid ||
+                form.formState.isSubmitting ||
+                api.ipWhitelist === form.watch("ipWhitelist")
+              }
+              type="submit"
+              loading={form.formState.isSubmitting}
+            >
+              Save
+            </Button>
+          </div>
+        ) : (
+          <div className="flex w-full gap-4 lg:justify-end lg:items-center ">
+            <a target="_blank" rel="noreferrer" href="https://cal.com/james-r-perkins/sales">
+              <Button
+                type="button"
+                variant="primary"
+                className="flex items-center justify-end gap-1 font-medium text-info-11 bg-info-4 leading-4 text-[13px] px-3 h-9 rounded-lg border border-info-5 hover:bg-info-3 w-[16rem] lg:w-[12rem]"
+              >
+                Upgrade to Enterprise
+              </Button>
+            </a>
+            <a
+              href="https://www.unkey.com/docs/apis/features/whitelist#ip-whitelisting"
+              target="_blank"
+              rel="noreferrer"
+              className="flex items-center justify-end"
+            >
+              <Button
+                type="button"
+                variant="ghost"
+                className="py-3 gap-1 font-medium text-accent-12 leading-4 text-[13px]"
```

Comment:
this custom styling does not fit our sizes
![CleanShot 2025-04-23 at 08 55 31@2x](https://github.com/user-attachments/assets/7f3bd09a-020e-41ba-b79f-ff3038e34f4f)